src/test/scala/org/apache/sysml/test/integration/applications/mlpipeline/LogisticRegressionSuite.scala

```diff
@@ -0,0 +1,147 @@
+    
+    //scored_systemml.select("prediction").collect() shouldEqual scored_spark.select("prediction").collect()
+  }
+
+  test("test logistic regression with mlpipeline"){
+    //Make sure system ml home set when run wrapper
+    val newsqlContext = new org.apache.spark.sql.SQLContext(sc);
+    import newsqlContext.implicits._
+    val training = sc.parallelize(Seq(
+	     LabeledDocument(0L, "a b c d e spark", 1.0),
+	     LabeledDocument(1L, "b d", 2.0),
+	     LabeledDocument(2L, "spark f g h", 1.0),
+	     LabeledDocument(3L, "hadoop mapreduce", 2.0),
+	     LabeledDocument(4L, "b spark who", 1.0),
+	     LabeledDocument(5L, "g d a y", 2.0),
+	     LabeledDocument(6L, "spark fly", 1.0),
+	     LabeledDocument(7L, "was mapreduce", 2.0),
+	     LabeledDocument(8L, "e spark program", 1.0),
+	     LabeledDocument(9L, "a e c l", 2.0),
+	     LabeledDocument(10L, "spark compile", 1.0),
+	     LabeledDocument(11L, "hadoop software", 2.0)))
+	     
+    val test = sc.parallelize(Seq(
+      Document(12L, "spark i j k"),
+      Document(13L, "l m n"),
+      Document(14L, "mapreduce spark"),
+      Document(15L, "apache hadoop")))
+      
+    val tokenizer = new Tokenizer().setInputCol("text").setOutputCol("words")
+    val hashingTF = new HashingTF().setNumFeatures(1000).setInputCol(tokenizer.getOutputCol).setOutputCol("features")
+    
+    
+    val lr_systemml = new LogisticRegression("log",sc)
+    val pipeline_systemml = new Pipeline().setStages(Array(tokenizer, hashingTF, lr_systemml))
+    val crossval_systemml = new CrossValidator().setEstimator(pipeline_systemml).setEvaluator(new BinaryClassificationEvaluator)
+    val paramGrid_systemml = new ParamGridBuilder().addGrid(hashingTF.numFeatures, Array(10, 100, 1000)).addGrid(lr_systemml.regParam, Array(0.1, 0.01)).build()
+    crossval_systemml.setEstimatorParamMaps(paramGrid_systemml)
+    crossval_systemml.setNumFolds(2)
+    val model_systemml = crossval_systemml.fit(training.toDF)
+    val scored_systemml = model_systemml.transform(test.toDF).orderBy("id")
+    
+    val lr_spark = new org.apache.spark.ml.classification.LogisticRegression("log")
+    val pipeline_spark = new Pipeline().setStages(Array(tokenizer, hashingTF, lr_spark))
+    val crossval_spark = new CrossValidator().setEstimator(pipeline_spark).setEvaluator(new BinaryClassificationEvaluator)
+    val paramGrid_spark = new ParamGridBuilder().addGrid(hashingTF.numFeatures, Array(10, 100, 1000)).addGrid(lr_systemml.regParam, Array(0.1, 0.01)).build()
+    crossval_spark.setEstimatorParamMaps(paramGrid_spark)
+    crossval_spark.setNumFolds(2)
+    val model_spark = crossval_spark.fit(training.map { x => LabeledDocument(x.id,x.text, x.label-1)}.toDF)
+    val scored_spark = model_spark.transform(test.toDF).orderBy("id")
+    
+    scored_systemml.select("prediction").collect() shouldEqual scored_spark.select(scored_spark("prediction")+1).collect()
```

Comment:
For these wrappers that are designed to be used as a direct replacement for Spark ML algorithms, I think that testing against those Spark ML algorithms would be appropriate.